compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingAttrs.td

```diff
@@ -63,8 +63,19 @@ def EncodingAttr :
     AttrParameter<"ArrayAttr", "element types of the user's operands">:$element_types,
     OptionalParameter<"ArrayAttr", "Indexing maps of the operation using this tensor">:$user_indexing_maps,
     OptionalParameter<"AffineMapAttr", "Indexing map that represents the broadcasting dims in the producer">:$bcast_map,
-    // TODO(hanchung): The round_dims_to parameter can be revisited. We explicitly map them to M,N,K dimension for now.
-    OptionalParameter<"DenseArrayAttr", "Values for padding M,N,K dimensions">:$round_dims_to
+    // TODO(hanchung): Deprecate the round_dims_to field when we plumb the layouts
+    // field through the whole stack. See https://github.com/iree-org/iree/issues/17924
+    // for details. Note that today we abuse the attribute to carry narrow
+    // matrix information. The end goal is deprecating the field and add a
+    // "iteration_space_size" field to describe the shape. It is useful to
+    // handle narrow matrix cases.
+    OptionalParameter<"DenseArrayAttr", "Values for padding M,N,K dimensions">:$round_dims_to,
+    OptionalParameter<"ArrayAttr", "An array of attributes that describes the "
+    "potential layouts on the device. It is an array because a device could "
+    "have several executable targets. Note that it can be any attribute with "
+    "encoding attribute interface implementation. The expectation of the field "
```

Comment:
I should probably rebase this on top of https://github.com/iree-org/iree/pull/19216, so I can explicitly use the `EncodingLayoutAttrInterface` term here. Does it sound better?

Reply:
I do fell like it will be great to combine the two (or rebase one on top of the other) and review them all together. The reason being one about the design and one is about implementation/application, so they share context.

Reply:
Let me rebase this on top of the other one. The PRs look relevant, but honestly I don't know how to write PR title and description if we combine them all together. I agree that the code is in a weird stage because there are no actual users. I should be able to add some unittests in upcoming PRs once I have them landed.